packages/contracts/contracts/RollupMerkleUtils.sol

```diff
@@ -0,0 +1,268 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+/* Internal Imports */
+import {DataTypes as dt} from "./DataTypes.sol";
+
+/*
+ * Merkle Tree Utilities for Rollup
+ */
+contract RollupMerkleUtils {
+    /* Structs */
+    // A partial merkle tree which can be updated with new nodes, recomputing the root
+    struct SparseMerkleTree {
+        // The root
+        bytes32 root;
+        uint height;
+        mapping (bytes32 => bytes32) nodes;
+    }
+
+    /* Fields */
+    // The default hashes
+    bytes32[160] public defaultHashes;
+    // A tree which is used in `update()` and `store()`
+    SparseMerkleTree public tree;
+
+    /**
+     * @notice Initialize a new SparseMerkleUtils contract, computing the default hashes for the sparse merkle tree (SMT)
+     */
+    constructor() public {
+        // Calculate & set the default hashes
+        setDefaultHashes();
+    }
+
+    /* Methods */
+    /**
+     * @notice Set default hashes
+     */
+    function setDefaultHashes() private {
+        // Set the initial default hash.
+        // TODO: Replace this hardcoded bytes value with keccak256(0x000...000) or similar.
+        defaultHashes[0] = 0x290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563;
```

Comment:
I  know from my tree stuff that this is the result of `keccack256(ZERO_BYTES32)` 😄 . Anyone  reason why this isn't calculated here, though?